SparcInstPrinter.c:361:17: error: ‘__builtin_strncpy’ specified bound 64 equals destination size [-Werror=stringop-truncation]

```
FAILED: subprojects/capstone-next/libcapstone.a.p/arch_Sparc_SparcInstPrinter.c.o 
gcc-12 -Isubprojects/capstone-next/libcapstone.a.p -I../subprojects/capstone-next/include -I../subprojects/capstone-next/include/capstone -fdiagnostics-color=always -fsanitize=address,undefined -fno-omit-frame-pointer -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O2 -g --std=gnu99 -Werror=sizeof-pointer-memaccess -fvisibility=hidden -DASAN=1 -DRZ_ASSERT_STDOUT=1 -Wno-cpp -ftrivial-auto-var-init=pattern -funsigned-char -fPIC -DCAPSTONE_X86_ATT_DISABLE_NO -DCAPSTONE_X86_REDUCE_NO -DCAPSTONE_USE_SYS_DYN_MEM -DCAPSTONE_DIET_NO -DCAPSTONE_HAS_ARM -DCAPSTONE_HAS_ARM64 -DCAPSTONE_HAS_M68K -DCAPSTONE_HAS_M680X -DCAPSTONE_HAS_MIPS -DCAPSTONE_HAS_POWERPC -DCAPSTONE_HAS_SPARC -DCAPSTONE_HAS_SYSZ -DCAPSTONE_HAS_X86 -DCAPSTONE_HAS_XCORE -DCAPSTONE_HAS_TMS320C64X -MD -MQ subprojects/capstone-next/libcapstone.a.p/arch_Sparc_SparcInstPrinter.c.o -MF subprojects/capstone-next/libcapstone.a.p/arch_Sparc_SparcInstPrinter.c.o.d -o subprojects/capstone-next/libcapstone.a.p/arch_Sparc_SparcInstPrinter.c.o -c ../subprojects/capstone-next/arch/Sparc/SparcInstPrinter.c
../subprojects/capstone-next/arch/Sparc/SparcInstPrinter.c: In function ‘Sparc_printInst’:
../subprojects/capstone-next/arch/Sparc/SparcInstPrinter.c:361:17: error: ‘__builtin_strncpy’ specified bound 64 equals destination size [-Werror=stringop-truncation]
  361 |                 strncpy(instr, mnem, sizeof(instr));
      |                 ^
cc1: all warnings being treated as errors
```

--- a/arch/Sparc/SparcInstPrinter.c
+++ b/arch/Sparc/SparcInstPrinter.c
@@ -20,10 +20,6 @@
 #define _CRT_SECURE_NO_WARNINGS
 #endif
 
-#if defined (WIN32) || defined (WIN64) || defined (_WIN32) || defined (_WIN64)
-#pragma warning(disable:28719)		// disable MSVC's warning on strncpy()
-#endif
-
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -358,8 +354,9 @@ void Sparc_printInst(MCInst *MI, SStream *O, void *Info)
 	mnem = printAliasInstr(MI, O, Info);
 	if (mnem) {
 		// fixup instruction id due to the change in alias instruction
-		strncpy(instr, mnem, sizeof(instr));
-		instr[sizeof(instr) - 1] = '\0';
+		unsigned cpy_len = sizeof(instr) < strlen(mnem) ? sizeof(instr) : strlen(mnem);
+		memcpy(instr, mnem, cpy_len);
+		instr[cpy_len - 1] = '\0';
 		// does this contains hint with a coma?
 		p = strchr(instr, ',');
 		if (p)

